client/internal/dns/server.go

```diff
@@ -131,7 +142,11 @@ func (s *DefaultServer) Initialize() (err error) {
 		}
 	}
 
-	s.hostManager, err = newHostManager(s.wgInterface)
+	if runtime.GOOS == "ios" {
```

Comment:
I am not sure in that make sense for this condition. You can call the newHostManager with s.wgInterface, manager arguments. The proper implementations of the host managers will handle the nil things.

Reply:
Agree